rocketpy/simulation/flight.py

```diff
@@ -1053,6 +726,573 @@ def __simulate(self, verbose):
+        """Process overshootable time nodes for parachutes, controllers, and sensors.
+
+        Parameters
+        ----------
+        phase : FlightPhase
+            The current flight phase.
+        phase_index : int
+            The index of the current phase.
+        node_index : int
+            The index of the current node.
+
+        Returns
+        -------
+        bool
+            True if a parachute was triggered and the simulation should break.
+        """
+        overshootable_nodes = self.TimeNodes()
+
+        overshootable_nodes.add_parachutes(
+            self.parachutes, self.solution[-2][0], self.t
+        )
+        overshootable_nodes.add_controllers(
+            self._controllers, self.solution[-2][0], self.t
+        )
+        overshootable_nodes.add_sensors(
+            self.rocket.sensors, self.solution[-2][0], self.t
+        )
+
+        # Add last time node (always skipped)
+        overshootable_nodes.add_node(self.t, [], [], [])
+
+        if len(overshootable_nodes) < 1:
+            return False  # Early exit
+
+        overshootable_nodes.sort()
+        overshootable_nodes.merge()
+
+        # Clear if necessary
+        if overshootable_nodes[0].t == phase.t and phase.clear:
+            overshootable_nodes[0].parachutes = []
+            overshootable_nodes[0].callbacks = []
+
+        # Feed overshootable time nodes trigger
+        interpolator = phase.solver.dense_output()
+        for overshootable_index, overshootable_node in self.time_iterator(
+            overshootable_nodes
+        ):
+            # Calculate state at node time
+            overshootable_node.y_sol = interpolator(overshootable_node.t)
+
+            # Check for parachute triggers
+            if self.__check_overshootable_parachute_triggers(
+                overshootable_node,
+                overshootable_nodes,
+                overshootable_index,
+                phase,
+                phase_index,
+                node_index,
+            ):
+                return True
+
+            # Process controllers at overshootable node
+            for controller in overshootable_node._controllers:
+                controller(
+                    overshootable_node.t,
+                    overshootable_node.y_sol,
+                    self.solution,
+                    self.sensors,
+                )
+
+            # Process sensors at overshootable node
+            if overshootable_node._component_sensors:
+                # Calculate u_dot for sensors at interpolated state
+                u_dot = phase.derivative(overshootable_node.t, overshootable_node.y_sol)
+                self.__measure_sensors(
+                    overshootable_node._component_sensors,
+                    u_dot,
+                    overshootable_node.t,
+                    overshootable_node.y_sol,
+                )
+        return False
+
+    def __check_overshootable_parachute_triggers(
```

Comment:
This method and the `__handle_parachute_triggers` are essentially identical (https://www.diffchecker.com/FYbfsMQf/) aside from the history rollback. They could be converted intro a single method with a boolean parameter to toggle the rollback.